### PR TITLE
Avoid dependency on system tzdata

### LINKED ISSUE #10211: Africa/Casablanca timezone missing
Hello,

Putting **Africa/Casablanca** as a timezone in both config.xml or in queries is not working.

Build : ClickHouse server version 20.3.5.21 (official build).

Regards

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Avoid dependency on system tzdata. This fixes loading of `Africa/Casablanca` timezone on CentOS 8. This fixes #10211

### CODE REVIEW

File: base/common/DateLUTImpl.cpp [hunk 1]
```diff
@@ -157,3 +160,80 @@ DateLUTImpl::DateLUTImpl(const std::string & time_zone_)
...
+        {
+            std::string name_replaced = name;
+            std::replace(name_replaced.begin(), name_replaced.end(), '/', '_');
+            std::replace(name_replaced.begin(), name_replaced.end(), '-', '_');
+
+            /// These are the names that are generated by "ld -r -b binary"
+            std::string symbol_name_data = "_binary_" + name_replaced + "_start";
+            std::string symbol_name_size = "_binary_" + name_replaced + "_size";
+
+            const void * sym_data = dlsym(RTLD_DEFAULT, symbol_name_data.c_str());
```
Comment (alesapin) [inline base/common/DateLUTImpl.cpp:220]: I haven't found any function for this, but maybe it's possible to get a symbol address without `dlsym`? It should be possible to lookup symbols in static ELF binaries.

Comment (alexey-milovidov) [reply base/common/DateLUTImpl.cpp:220]: Yes, it's also possible: we can open ELF file and navigate its sections, see ELF.h/cpp.
But I don't know any convenient methods for it.

See also: https://stackoverflow.com/questions/34888385/dlsym-like-functionality-for-non-dynamically-loaded-code

Comment (alexey-milovidov) [reply base/common/DateLUTImpl.cpp:220]: Less hacky way is to declare every needed symbol as `extern const char *` variable and access directly. But we access them by their names. Nevertheless, we can create a file with index of these symbols at build time and compile it.

**Comment by alexey-milovidov** [comment]
Side note: multiple time zones are identical. We can coalesce them at build time.
But as they are small it's ok as is.

**Review by alesapin — APPROVED** [review]

**Pull request merged**